tests/unit/notification/utils/config.js

```diff
@@ -0,0 +1,177 @@
+            queueConfigurations: [
+                {
+                    events: ['s3:ObjectCreated:Copy'],
+                    queueArn: 'q3',
+                    filter: {},
+                    id: 'config4',
+                },
+            ],
+        },
+    },
+];
+
+const tests = [
+    {
+        desc: 'pass if the event matches a bucket notification configuration',
+        entry: {
+            type: 's3:ObjectCreated:Copy',
+            bucket: 'bucket3',
+            key: 'test.png',
+        },
+        pass: true,
+    },
+    {
+        desc: 'pass if the object name prefix matches the configuration',
+        entry: {
+            type: 's3:ObjectCreated:Put',
+            bucket: 'bucket1',
+            key: 'test.png',
+        },
+        pass: true,
+    },
+    {
+        desc: 'pass if the object name suffix matches the configuration',
+        entry: {
+            type: 's3:ObjectCreated:Copy',
+            bucket: 'bucket1',
+            key: 'test.png',
+        },
+        pass: true,
+    },
+    {
+        desc: 'fail if the object bucket has no configuration setup',
+        entry: {
+            type: 's3:ObjectCreated:Put',
+            bucket: 'bucket10',
+            key: 'test.png',
+        },
+        pass: false,
+    },
+    {
+        desc: 'fail if the event type does not match the configuration',
+        entry: {
+            type: 's3:ObjectCreated:Post',
+            bucket: 'bucket1',
+            key: 'test.png',
+        },
+        pass: false,
+    },
+    {
+        desc: 'fail if the object name does not match configuration prefix',
+        entry: {
+            type: 's3:ObjectCreated:Put',
+            bucket: 'bucket1',
+            key: 'one.png',
+        },
+        pass: false,
+    },
+    {
+        desc: 'fail if the object name does not match configuration suffix',
+        entry: {
+            type: 's3:ObjectCreated:Copy',
+            bucket: 'bucket1',
+            key: 'test.jpg',
+        },
+        pass: false,
+    },
+];
+
+describe('Notification configuration util', () => {
+    describe('Transform', () => {
+        it('should transform an array of configs to a Map', () => {
+            const configMap = notifConfigUtil.transform(testConfigs);
+            assert(configMap instanceof Map);
+            testConfigs.forEach(config => {
+                const bucket = config.bucket;
+                assert(configMap.has(bucket));
+                const currentConfig = configMap.get(bucket);
+                const bucketConfig
+                    = testConfigs.filter(c => c.bucket === bucket);
+                assert.strictEqual(bucketConfig.length, currentConfig.length);
+            });
+        });
+    });
+
+    describe('ValidateEntry', () => {
+        const configMap = notifConfigUtil.transform(testConfigs);
+
+        tests.forEach(test => {
+            it(`should ${test.desc}`, () => {
+                const bckt = test.entry.bucket;
```

Comment:
why not simply: `const { bucket, key, type } = test.entry;`

Reply:
Was struggling to fit the variable names with column restrictions. Changed it.

Reply:
Don't hesitate to get rid of the current drastic column restriction by updating the dependency to eslint-config-scality, it will definitely help in the long run to avoid reducing readability to comply with the linter rules.